frontend/src/citizen-frontend/utils/duplicated-child-utils.ts

```diff
@@ -0,0 +1,23 @@
+// SPDX-FileCopyrightText: 2017-2023 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+import { Translations } from 'citizen-frontend/localization'
+import { Child } from 'lib-common/generated/api-types/children'
+import { ReservationChild } from 'lib-common/generated/api-types/reservations'
+
+export const toDuplicatedChildIds = (
+  children: Array<Child | ReservationChild>
+) =>
+  children
+    .filter((child) => child.duplicateOf !== null)
+    .flatMap((child) => [child.id, child.duplicateOf as string])
```

Comment:
Just FYI, you can use `flatMap` for filtering in one call. The types will also be correct automatically:

```suggestion
  children
    .flatMap((child) => child.duplicateOf !== null ? [child.id, child.duplicateOf] : [])
```